webapp/channels/src/components/gif_picker/components/gif_picker_search.tsx

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {ChangeEvent, memo, useEffect, useState} from 'react';
+import {useIntl} from 'react-intl';
+import {useSelector} from 'react-redux';
+import tinycolor from 'tinycolor2';
+
+import {getTheme} from 'mattermost-redux/selectors/entities/preferences';
+
+import {calculateContrastRatio} from 'utils/colors';
+
+import giphyWhiteImage from 'images/gif_picker/powered-by-giphy-white.png';
+import giphyBlackImage from 'images/gif_picker/powered-by-giphy-black.png';
+
+interface Props {
+    value: string;
+    onChange: (value: string) => void;
+}
+
+function GifPickerSearch(props: Props) {
+    const theme = useSelector(getTheme);
+
+    const [useDarkLogo, setDarkLogo] = useState(false);
+
+    const {formatMessage} = useIntl();
+
+    const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
```

Comment:
handleChange is used as a prop, so should use a `useCallback`. That being said, the element that is sent to is a basic element, so not sure the gain there.

I prefer taking the "purist way" and make it a `useCallback`, but 0/5.

Reply:
changed to callback